### PR TITLE
[SQL] Add 'AuthenticationMetadata' option to New and Set AzSqlInstance cmdlets

### PR DESCRIPTION
**Pull request by krivi37**
## Description

This PR adds option to configure which metadata should be used for authenticating users which were synced from on-prem Active Directory domains. The authentication metadata has 3 options: `AzureAD`, `Paired` and `Windows`. 
These options dictate which metadata in managed instance will be used for on-prem synced users for authentication:
  - `AzureAD` - all on-prem synced users must have Microsoft Entra (AAD) logins/users present in metadata in order to be able to login to the managed instance.
  - `Paired` - on-prem synced users will use metadata depending on the type of the authentication they use. If they try to connect using Microsoft Entra authentication, then they must have Microsoft Entra logins/users. In case they try to connect using Windows authentication, they must have Windows logins/users present in the metadata.
  - `Windows` - all on-prem synced users must have Windows logins/users present in metadata in order to be able to login to the managed instance.

These options are exposed by adding `AuthenticationMetadata` option to the `New-AzSqlInstance` and `Set-AzSqlInstance` cmdlets.

The default authentication metadata option is `AzureAD`.

For more information, please refer to: https://learn.microsoft.com/en-us/azure/azure-sql/managed-instance/native-windows-principals?view=azuresql

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [X] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Sql/Sql/ChangeLog.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
 ## Version 5.1.0
 * Added cross-subscription support for `Copy-AzSqlInstanceDatabase`, `Move-AzSqlInstanceDatabase`
 * Added new parameter SecondaryType to `Add-AzSqlDatabaseFromFailoverGroup`
+* Added new parameter AuthenticationMetadata to `New-AzSqlInstance` and `Set-AzSqlInstance`
```
Comment (dolauli) [inline src/Sql/Sql/ChangeLog.md:30]: Please move the change log to top of **Upcoming Release**

**Review by dolauli — CHANGES_REQUESTED** [review]

File: src/Sql/Sql/ChangeLog.md (hunk 1 above)
Comment (krivi37) [reply src/Sql/Sql/ChangeLog.md]: Hmm it seems Git merged it badly after pulling in the latest changes from main branch...
I've moved it to Upcoming release section

**Review by dolauli — APPROVED** [review]
LGTM

**Pull request merged**